packages/compass-generative-ai/src/atlas-ai-service.ts

```diff
@@ -395,6 +467,60 @@ export class AtlasAiService {
     );
   }
 
+  /**
+   * @returns {MockDataSchemaResponse} which contains faker.js mappings used to produce a
+   * faker.js factory function for the purposes of generating mock document data.
+   *
+   * @throws {AtlasAiServiceApiResponseParseError} when the response cannot be parsed into the expected schema.
+   */
+  async getMockDataSchema(
+    input: MockDataSchemaRequest,
+    connectionInfo: ConnectionInfo
+  ): Promise<MockDataSchemaResponse> {
+    const { collectionName, databaseName } = input;
+    let schema = input.schema;
+
+    const url = `${this.getUrlForEndpoint('mock-data-schema', connectionInfo)}`;
+
+    if (!input.includeSampleValues) {
+      const newSchema: Record<
+        string,
+        Omit<MockDataSchemaRawField, 'sampleValues'>
+      > = {};
+      for (const [k, v] of Object.entries(schema)) {
+        newSchema[k] = { type: v.type, probability: v.probability };
+      }
+      schema = newSchema;
+    }
+
+    const res = await this.atlasService.authenticatedFetch(url, {
+      method: 'POST',
+      body: JSON.stringify({
+        collectionName,
+        databaseName,
+        schema,
+      }),
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+    });
+
+    try {
+      const data = await res.json();
+      return MockDataSchemaResponseShape.parse(data);
+    } catch {
+      this.logger.log.error(
+        mongoLogId(1_001_000_311),
+        'AtlasAiService',
+        'Failed to parse mock data schema response with expected schema'
+      );
```

Comment:
Should we catch the error object and pass it along to the logger?

Reply:
resolved. 👍🏼 I passed the relevant error attributes